qiskit_algorithms/optimizers/scipy_optimizer.py

```diff
@@ -76,6 +76,20 @@ def __init__(
         self._max_evals_grouped = max_evals_grouped
         self._kwargs = kwargs
 
+        # Initialize bounds and re-allocate if definition in options or kwargs
+        self._bounds: list[tuple[float, float]] | None = None
```

Comment:
Since I removed the logic that checks for compatibility (turned out not to be robust enough to justify including it) this is also not necessary, and I will remove it in the upcoming commit.